packages/block-library/src/breadcrumbs/index.php

```diff
@@ -157,19 +157,53 @@ function block_core_breadcrumbs_maybe_add_paged( $breadcrumb_items ) {
 	return $breadcrumb_items;
 }
 
+/**
+ * Generates post type archive breadcrumb item.
+ *
+ * Returns the post type archive link item if the post type has archive enabled,
+ * otherwise returns null.
+ *
+ * @since 6.9.0
+ *
+ * @param string $post_type The post type name.
+ *
+ * @return string|null The archive breadcrumb item HTML, or null if not available.
+ */
+function block_core_breadcrumbs_get_post_type_archive_item( $post_type ) {
+	$post_type_object = get_post_type_object( $post_type );
+	if ( $post_type_object && $post_type_object->has_archive ) {
+		$archive_link = get_post_type_archive_link( $post_type );
+		if ( $archive_link ) {
+			return sprintf(
+				'<a href="%s">%s</a>',
+				esc_url( $archive_link ),
+				esc_html( $post_type_object->labels->name )
+			);
+		}
+	}
+	return null;
+}
+
 /**
  * Generates breadcrumb items from hierarchical post type ancestors.
  *
  * @since 6.9.0
  *
  * @param int    $post_id   The post ID.
+ * @param string $post_type The post type name.
  *
  * @return array Array of breadcrumb HTML items.
  */
-function block_core_breadcrumbs_get_hierarchical_post_type_breadcrumbs( $post_id ) {
+function block_core_breadcrumbs_get_hierarchical_post_type_breadcrumbs( $post_id, $post_type ) {
```

Comment:
It feels non-ergonomic that we also have to pass the post type explicitly. I'd rather grab it from the `$post_id`:

```php
$post_type = get_post_type( $post_id );
```

In theory, it should be free to do that since the post object is already in the `WP_Query`.

Reply:
I would agree if we had it inline, or if we didn't have it loaded from the database. But it should already be loaded, and we don't have it inline in that function either - we pass it between functions, and to do so we need to add an extra argument. This makes it less ergonomic, which is something to cosnider when designing functions and APIs. 

That being said, I don't feel strongly about it, I don't think it's a major deal.